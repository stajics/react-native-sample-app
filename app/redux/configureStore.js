import { applyMiddleware, createStore, compose } from 'redux';
// middleware
import createLogger from 'redux-logger';
import thunk from 'redux-thunk';
import authenticatedApiCall from './middleware/authenticatedApiCall';
import apiCall from './middleware/apiCall';
import rootReducer from './rootReducer';
// import sessioning from './utils/middleware/sessioning';

const logger = createLogger();
const middleware = [thunk, apiCall, authenticatedApiCall, logger];

const configureStore = (initialState = undefined) => {
  const store = compose(applyMiddleware(...middleware))(createStore)(rootReducer, initialState);
  return store;
};

export default configureStore;
