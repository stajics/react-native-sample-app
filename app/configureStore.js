import { applyMiddleware, createStore, compose } from 'redux';
// middleware
import createLogger from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './rootReducer';
// import sessioning from './utils/middleware/sessioning';

const logger = createLogger();
// const middleware = [thunk, sessioning, logger];
const middleware = [thunk, logger];

const configureStore = () => {
  const store = compose(applyMiddleware(...middleware))(createStore)(rootReducer);
  return store;
};

export default configureStore;
