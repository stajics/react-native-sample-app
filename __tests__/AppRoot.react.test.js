import 'react-native';
import React from 'react';
import AppRoot from '../app/AppRoot.react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'; // eslint-disable-line

// store
import { Provider } from 'react-redux'; // eslint-disable-line
import configureStore from '../app/redux/configureStore';

const store = configureStore();

it('renders correctly', () => {
  const appRoot = renderer.create(
    <Provider store={store}>
      <AppRoot />
    </Provider>
  ).toJSON();
  expect(appRoot).toMatchSnapshot();
});
