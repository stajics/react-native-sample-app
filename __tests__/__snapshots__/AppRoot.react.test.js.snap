exports[`test renders correctly 1`] = `
<View
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        undefined,
      ]
    }>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }>
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }>
              Sample App
            </Text>
            <Image
              source={
                Object {
                  "uri": "http://image.flaticon.com/teams/new/1-freepik.jpg",
                }
              }
              style={
                Object {
                  "height": 200,
                  "width": 200,
                }
              } />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
